whylog_vim/gui/__init__.py

```diff
@@ -0,0 +1,105 @@
+from whylog import LineSource, FrontInput
+
+from mock import MagicMock, patch
+import os.path
+import vim
+from whylog_vim.input_reader.teacher_reader import get_button_name, filter_comments
+from whylog_vim.consts import WindowTypes, WindowSizes
+from whylog_vim.gui.vim_commander import VimCommander
+from whylog_vim.gui.windows import WhylogWindowManager
+from whylog_vim.gui.files_manager import FilesManager
+
+
+class VimEditor(object):
+
+    def __init__(self):
+        self.window_manager = WhylogWindowManager()
+
+    def get_input_content(self):
+        return filter_comments(
+            self.window_manager.get_window_content(WindowTypes.INPUT))
+
+    def create_case_window(self, default_input=None):
+        self.window_manager.create_window(WindowTypes.CASE, default_input)
+
+    def create_input_window(self, default_input=None):
+        self.window_manager.create_window(WindowTypes.INPUT, default_input, modifiable=True)
+
+    def create_teacher_window(self, default_input=None):
+        self.window_manager.create_window(WindowTypes.TEACHER, default_input)
+
+    def create_query_window(self, default_input=None):
+        self.window_manager.create_window(WindowTypes.QUERY, default_input, False, WindowSizes.QUERY_WINDOW)
+
+    def go_to_file(self, filename, offset=1):
+        FilesManager.go_to_file(filename, offset)
+
+    def set_query_output(self, content):
+        self.window_manager.set_content(WindowTypes.QUERY, content)
+
+    def set_teacher_output(self, content):
+        self.window_manager.set_content(WindowTypes.TEACHER, content)
+
+    def is_any_whylog_window_open(self):
+        return not self.window_manager.are_windows_closed()
+
+    def close_window(self, filename):
+        if not FilesManager.is_file_open(filename):
+            VimCommand.close_current_window()
+
+    def get_button_name(self):
+        return get_button_name(VimCommander.get_current_line(), VimCommander.get_column())
+
+    def get_front_input(self):
+        """
+        This method returns Front Input object of the line where cursor is,
+        """
+        filename = self.get_current_filename()
+        host = 'localhost'
```

Comment:
I suppose it is temoporary. Change it to finding proper host or write TODO comment